rdflib/plugins/sparql/operators.py

```diff
@@ -998,6 +999,17 @@ def datetime(e):
     return e.toPython()
 
 
+def date(e) -> py_datetime.date:
+    if not isinstance(e, Literal):
+        raise SPARQLError("Non-literal passed as date: %r" % e)
+    if e.datatype not in (XSD.date, XSD.dateTime):
+        raise SPARQLError("Literal with wrong datatype passed as date: %r" % e)
+    result = e.toPython()
+    if isinstance(result, py_datetime.datetime):
+        return result.date()
+    return result
```

Comment:
Why is it neccessary to convert datetime to date, if `.year()`, `.month()` and `.day()` are available on datetime as well?

Reply:
I preferred to have a consistent return type and always return a `date`